src/logic/map_objects/tribes/ship.cc

```diff
@@ -2361,7 +2362,13 @@ void Ship::draw_healthbar(const EditorGameBase& egbase,
 		while (n_cols * n_rows < bonus) {
 			++n_cols;
 		}
```

Comment:
OK, so this is supposed to keep columns as wide as the healthbar, until it would require more than `kMaxRows`. Above that, keep rows at `kMaxRows` and allow increasing columns as much as required.

I suggest making it more explicit:
```
constexpr unsigned kDefaultColumns = 2 * kShipHealthBarWidth / kBonusIconSize;
constexpr unsigned kMaxBonusWithDefaultColumns = kDefaultColumns * kMaxRows;

unsigned n_cols = bonus > kMaxBonusWithDefaultColumns ?
   bonus / (kMaxRows - 1) :
   std::min(kMaxBonusWithDefaultColumns, bonus);
unsigned n_rows = bonus / n_cols;
if (n_cols * n_rows < bonus) {
	++n_rows;
}
const unsigned last_row_cols = …
```

I think this would be both more readable and less error-prone.

Reply:
`n_cols` could be further complicated to make exact multiples of 16 look neat, but I don't think it's worth the trouble.

Reply:
(this was supposed to be for lines 2356-2364)